numpy/distutils/fcompiler/environment.py

```diff
@@ -70,6 +72,8 @@ def _get_var(self, name, conf_desc):
                 else:
                     var = envvar_contents
         if confvar is not None and self._conf:
+            # FIXME: conf shall do the conversion immediately after the key is parsed.
+            # postpartum conversion will not always catch the right cases.
             var = self._conf.get(confvar, (None, var))[1]
         if convert is not None:
             var = convert(var)
```

Comment:
Lines 74 to here look broken to me in the same way - I think it should be:
```python
if confvar is not None and self._conf:
    try:
        source, confvar_contents = self._conf[confvar]
    except KeyError:
        pass
    else:
        var = convert(confvar_contents)
```

Otherwise flags in `distutils.cfg` will be broken just like your env vars

Reply:
I see, _conf is just a simple dict. Then we shall implement the split logic here.

Reply:
Importantly, it's a dict loaded from a `cfg` file so I believe all its "contents" elements are strings